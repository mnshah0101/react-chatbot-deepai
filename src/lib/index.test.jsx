import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, test } from 'vitest'
import ChatBot from './index'

describe('HelloWorld', () => {
  test('HelloWorld component renders correctly', () => {
    const component = renderer.create(
       <ChatBot OPEN_AI_KEY="hello" model="gpt-4"
    max_tokens={150} chatbot_prompt='You are a chatbot assistant on the Hermes Group Website. We are an AI and software development company out of Richmond, Virginia that specializes in helping small businesses with creating websites and software development. It was made by college students for local businesses and help them with their online presence. Pricing will be discussed with the client and will be based on the complexity of the project. For specific questions that you cannot answer, ask the user to contact us at our email: hermesdevelopergroup@gmail.com. Please keep responses short, under 50 words.'
    headerText='Hermes Group Chatbot'
    placeholderText='Type a message...'
    buttonText='Chat with us'
     />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('The greetee prop works', () => {
    const component = renderer.create(
      <ChatBot OPEN_AI_KEY="hello" model="gpt-4"
    max_tokens={150} chatbot_prompt='You are a chatbot assistant on the Hermes Group Website. We are an AI and software development company out of Richmond, Virginia that specializes in helping small businesses with creating websites and software development. It was made by college students for local businesses and help them with their online presence. Pricing will be discussed with the client and will be based on the complexity of the project. For specific questions that you cannot answer, ask the user to contact us at our email: hermesdevelopergroup@gmail.com. Please keep responses short, under 50 words.'
    headerText='Hermes Group Chatbot'
    placeholderText='Type a message...'
    buttonText='Chat with us'
     />
    )

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
