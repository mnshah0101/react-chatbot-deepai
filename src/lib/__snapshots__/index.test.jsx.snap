// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld > HelloWorld component renders correctly 1`] = `
<div
  className="chatcontainer"
>
  <div
    className="chatbotbutton"
  >
    <button
      onClick={[Function]}
    >
      <span
        className="button_text"
      >
        Chat with us
      </span>
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2.678 11.894a1 1 0 0 1 .287.801 10.97 10.97 0 0 1-.398 2c1.395-.323 2.247-.697 2.634-.893a1 1 0 0 1 .71-.074A8.06 8.06 0 0 0 8 14c3.996 0 7-2.807 7-6 0-3.192-3.004-6-7-6S1 4.808 1 8c0 1.468.617 2.83 1.678 3.894zm-.493 3.905a21.682 21.682 0 0 1-.713.129c-.2.032-.352-.176-.273-.362a9.68 9.68 0 0 0 .244-.637l.003-.01c.248-.72.45-1.548.524-2.319C.743 11.37 0 9.76 0 8c0-3.866 3.582-7 8-7s8 3.134 8 7-3.582 7-8 7a9.06 9.06 0 0 1-2.347-.306c-.52.263-1.639.742-3.468 1.105z"
        />
      </svg>
    </button>
  </div>
  <div
    className="closedChat"
  >
    <div>
      <div>
        <div
          className="react-chatbot-kit-chat-container"
        >
          <div
            className="react-chatbot-kit-chat-inner-container"
          >
            <div
              className="react-chatbot-kit-chat-header"
            >
              Hermes Group Chatbot
            </div>
            <div
              className="react-chatbot-kit-chat-message-container"
            >
              <div
                style={
                  {
                    "paddingBottom": "15px",
                  }
                }
              />
            </div>
            <div
              className="react-chatbot-kit-chat-input-container"
            >
              <form
                className="react-chatbot-kit-chat-input-form"
                onSubmit={[Function]}
              >
                <input
                  className="react-chatbot-kit-chat-input"
                  onChange={[Function]}
                  placeholder="Type a message..."
                  value=""
                />
                <button
                  className="react-chatbot-kit-chat-btn-send"
                  style={
                    {
                      "backgroundColor": "",
                    }
                  }
                >
                  <svg
                    className="react-chatbot-kit-chat-btn-send-icon"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
                    />
                  </svg>
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HelloWorld > The greetee prop works 1`] = `
<div
  className="chatcontainer"
>
  <div
    className="chatbotbutton"
  >
    <button
      onClick={[Function]}
    >
      <span
        className="button_text"
      >
        Chat with us
      </span>
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2.678 11.894a1 1 0 0 1 .287.801 10.97 10.97 0 0 1-.398 2c1.395-.323 2.247-.697 2.634-.893a1 1 0 0 1 .71-.074A8.06 8.06 0 0 0 8 14c3.996 0 7-2.807 7-6 0-3.192-3.004-6-7-6S1 4.808 1 8c0 1.468.617 2.83 1.678 3.894zm-.493 3.905a21.682 21.682 0 0 1-.713.129c-.2.032-.352-.176-.273-.362a9.68 9.68 0 0 0 .244-.637l.003-.01c.248-.72.45-1.548.524-2.319C.743 11.37 0 9.76 0 8c0-3.866 3.582-7 8-7s8 3.134 8 7-3.582 7-8 7a9.06 9.06 0 0 1-2.347-.306c-.52.263-1.639.742-3.468 1.105z"
        />
      </svg>
    </button>
  </div>
  <div
    className="closedChat"
  >
    <div>
      <div>
        <div
          className="react-chatbot-kit-chat-container"
        >
          <div
            className="react-chatbot-kit-chat-inner-container"
          >
            <div
              className="react-chatbot-kit-chat-header"
            >
              Hermes Group Chatbot
            </div>
            <div
              className="react-chatbot-kit-chat-message-container"
            >
              <div
                className="react-chatbot-kit-chat-bot-message-container"
              >
                <div
                  className="react-chatbot-kit-chat-bot-avatar"
                >
                  <div
                    className="react-chatbot-kit-chat-bot-avatar-container"
                  >
                    <p
                      className="react-chatbot-kit-chat-bot-avatar-letter"
                    >
                      B
                    </p>
                  </div>
                </div>
                <div
                  className="react-chatbot-kit-chat-bot-message"
                  style={
                    {
                      "backgroundColor": "",
                    }
                  }
                >
                  <div
                    className="chatbot-loader-container"
                  >
                    <svg
                      height="21px"
                      id="dots"
                      version="1.1"
                      viewBox="0 0 132 58"
                      width="50px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="none"
                        stroke="none"
                      >
                        <g
                          fill="#fff"
                          id="chatbot-loader"
                        >
                          <circle
                            cx="25"
                            cy="30"
                            id="chatbot-loader-dot1"
                            r="13"
                          />
                          <circle
                            cx="65"
                            cy="30"
                            id="chatbot-loader-dot2"
                            r="13"
                          />
                          <circle
                            cx="105"
                            cy="30"
                            id="chatbot-loader-dot3"
                            r="13"
                          />
                        </g>
                      </g>
                    </svg>
                  </div>
                  <div
                    className="react-chatbot-kit-chat-bot-message-arrow"
                    style={
                      {
                        "borderRightColor": "",
                      }
                    }
                  />
                </div>
              </div>
              <div
                style={
                  {
                    "paddingBottom": "15px",
                  }
                }
              />
            </div>
            <div
              className="react-chatbot-kit-chat-input-container"
            >
              <form
                className="react-chatbot-kit-chat-input-form"
                onSubmit={[Function]}
              >
                <input
                  className="react-chatbot-kit-chat-input"
                  onChange={[Function]}
                  placeholder="Type a message..."
                  value=""
                />
                <button
                  className="react-chatbot-kit-chat-btn-send"
                  style={
                    {
                      "backgroundColor": "",
                    }
                  }
                >
                  <svg
                    className="react-chatbot-kit-chat-btn-send-icon"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
                    />
                  </svg>
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
